engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/marshall/ProcessMarshallerSpec.scala

```diff
@@ -91,18 +105,62 @@ class ProcessMarshallerSpec extends FlatSpec with Matchers with OptionValues wit
         |}
       """.stripMargin
 
-    inside(ProcessMarshaller.fromJson(withAdditionalFields)) { case Valid(process) =>
+    inside(ProcessMarshaller.fromJson(withoutAdditionalFields)) { case Valid(process) =>
       process.metaData.id shouldBe "custom"
       process.metaData.additionalFields shouldBe None
       process.nodes.head.data.additionalFields shouldBe None
     }
+  }
 
-    inside(ProcessMarshaller.fromJson(withoutAdditionalFields)) { case Valid(process) =>
+  it should "not marshall custom process additional fields" in {
+    val process = EspProcessBuilder
+      .id("process1")
+      .additionalFields(TestProcessAdditionalFields(field1 = "abc", field2 = 3))
+      .exceptionHandler()
+      .source("a", "")
+      .processorEnd("d", "dService", "p1" -> "expr3")
+
+    val result = ProcessMarshaller.toJson(process, PrettyParams.spaces2)
+
+    result should include (
+      """
+        |  "metaData" : {
+        |    "id" : "process1",
+        |    "typeSpecificData" : {
+        |      "type" : "StreamMetaData"
+        |    }
+        |  },""".stripMargin)
+  }
+
+  // TODO: There is no way to create a node with additional fields.
+
+  it should "unmarshall and omit custom additional fields" in {
+    val withAdditionalFields =
+      """
+        |{
+        |    "metaData" : {
+        |        "id": "custom",
+        |         "typeSpecificData": { "type" : "StreamMetaData", "parallelism" : 2 },
+        |         "additionalFields": { "custom" : "value" }
+        |    },
+        |    "exceptionHandlerRef" : { "parameters" : [ { "name": "errorsTopic", "expression": { "language": "spel", "expression": "error.topic" }}]},
+        |    "nodes" : [
+        |        {
+        |            "type" : "Source",
+        |            "id" : "start",
+        |            "ref" : { "typ": "kafka-transaction", "parameters": [ { "name": "topic", "expression": { "language": "spel", "expression": "in.topic" }}]},
+        |            "additionalFields": { "custom": "value" }
+        |        }
+        |    ]
+        |}
+      """.stripMargin
+
+    inside(ProcessMarshaller.fromJson(withAdditionalFields)) { case Valid(process) =>
       process.metaData.id shouldBe "custom"
       process.metaData.additionalFields shouldBe None
-      process.nodes.head.data.additionalFields shouldBe None
+      process.nodes should have size 1
+      process.nodes.head.data.additionalFields shouldBe Some(NodeAdditionalFields(description = None))
```

Comment:
would expect it to be None; nonetheless, it's ok.

Reply:
Yeah, it looks strange. I need to handle empty fields manually so it's the same for process additional fields.